scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -1727,3 +1727,178 @@ def test_len_and_bool():
+            rot_alt = Rotation.from_davenport(ax, order, 
+                                              angles_alt)
+                                              
+            assert_allclose(rot.as_quat(), rot_alt.as_quat())
+
+
+def test_as_davenport_degenerate():
+    # Since we cannot check for angle equality, we check for rotation matrix
+    # equality
+    rnd = np.random.RandomState(0)
+    n = 5
+    angles = np.empty((n, 3))
+
+    # symmetric sequences
+    angles[:, 0] = rnd.uniform(low=-np.pi, high=np.pi, size=(n,))
+    angles_middle = [rnd.choice([0, np.pi]) for i in range(n)]
+    angles[:, 2] = rnd.uniform(low=-np.pi, high=np.pi, size=(n,))
+    lambdas = rnd.uniform(low=0, high=np.pi, size=(5,))
+
+    e1 = np.array([1, 0, 0])
+    e2 = np.array([0, 1, 0])
+
+    for lamb in lambdas:
+        with pytest.warns(UserWarning, match="Gimbal lock"):
+            # Intrinsic
+            ax = [e1, e2, Rotation.from_rotvec(lamb*e2).apply(e1)]
+            angles[:, 1] = angles_middle - lamb
+            rot = Rotation.from_davenport(ax, 'intrinsic', angles)
+            mat_expected = rot.as_matrix()
+            angles_dav = rot.as_davenport(ax, 'intrinsic')
+            mat_estimated = Rotation.from_davenport(
+                ax, 'intrinsic', angles_dav).as_matrix()
+            assert_array_almost_equal(mat_expected, mat_estimated)
+
+        with pytest.warns(UserWarning, match="Gimbal lock"):
+            # Extrinsic
+            ax = [Rotation.from_rotvec(lamb*e2).apply(e1), e2, e1]
+            angles[:, 1] = angles_middle - lamb
+            rot = Rotation.from_davenport(ax, 'extrinsic', angles)
+            mat_expected = rot.as_matrix()
+            angles_dav = rot.as_davenport(ax, 'extrinsic')
+            mat_estimated = Rotation.from_davenport(
+                ax, 'extrinsic', angles_dav).as_matrix()
+            assert_array_almost_equal(mat_expected, mat_estimated)
+
+
+def test_compare_from_davenport_from_euler():
+
+    def basis_vec(axis):
+        if axis == 'x':
+            return [1, 0, 0]
+        elif axis == 'y':
+            return [0, 1, 0]
+        elif axis == 'z':
+            return [0, 0, 1]
+
+    rnd = np.random.RandomState(0)
+    n = 100
+    angles = np.empty((n, 3))
+
+    # symmetric sequences
+    angles[:, 0] = rnd.uniform(low=-np.pi, high=np.pi, size=(n,))
+    angles[:, 1] = rnd.uniform(low=0, high=np.pi, size=(n,))
+    angles[:, 2] = rnd.uniform(low=-np.pi, high=np.pi, size=(n,))
+    for order in ['extrinsic', 'intrinsic']:
+        for seq_tuple in permutations('xyz'):
+            seq = ''.join([seq_tuple[0], seq_tuple[1], seq_tuple[0]])
+            ax = [basis_vec(i) for i in seq]
+            if order == 'intrinsic':
+                seq = seq.upper()
+            eul = Rotation.from_euler(seq, angles)
+            dav = Rotation.from_davenport(ax, order, angles)
+            assert_allclose(eul.as_quat(), dav.as_quat(), atol=0, rtol=1e-9)
```

Comment:
Maybe better compare matrices as you did previously? Basing on that matrices are guaranteed to be unique.